pkg/log/log.go

```diff
@@ -384,7 +385,9 @@ func matchesPod(pod *corev1.Pod, options *Options) bool {
 }
 
 func matchesContainer(pod *corev1.Pod, container corev1.Container, options *Options) bool {
-	if options == nil || options.ContainerReplica == "" {
+	if options != nil && !options.IncludeProxyContainers && (container.Name == "linkerd-proxy" || container.Name == "linkerd-init") {
```

Comment:
Instead of hard coding these names we should check the containers against the "container spec". On each pod there is an annotation that has the v1.Container struct encoded in it.  The valid containers are, first, the value of `pod.Label[labels.AcornContainerName]` and then each key name in the Sidecars fields from the v1.Container.  An example of decoding this struct can be seen here.

https://github.com/acorn-io/acorn/blob/d12aeb70639466bad3396d7ba8b1e9dffaa05233/pkg/server/registry/apigroups/acorn/containers/translator.go#L95